src/helper_functions.rs

```diff
@@ -0,0 +1,200 @@
+use crate::RPSGame;
+use gstd::{exec, msg, prelude::*, ActorId};
+use io::*;
+
+impl RPSGame {
+    pub(crate) fn change_stage_by_timeout_if_needed(&mut self) {
+        let end_time = self.current_stage_start_timestamp
+            + match self.stage {
+                GameStage::Preparation => self.game_config.entry_timeout,
+                GameStage::InProgress(_) => self.game_config.move_timeout,
+                GameStage::Reveal(_) => self.game_config.reveal_timeout,
+            };
+
+        if end_time < exec::block_timestamp() {
+            match self.stage.clone() {
+                GameStage::Preparation => self.handle_preparation_timout(),
+                GameStage::InProgress(desctription) => self.handle_moves_timout(&desctription),
+                GameStage::Reveal(desctription) => self.handle_reveal_timout(&desctription),
+            }
```

Comment:
Poorly designed API forces you to clone `stage` despite the cloning could be avoided. Many typos too.
```suggestion
            match &self.stage {
                GameStage::Preparation => self.handle_preparation_timeout(),
                GameStage::InProgress(description) => self.handle_moves_timeout(description),
                GameStage::Reveal(description) => self.handle_reveal_timeout(description),
            }
```